reference/functions/processexists.markdown

```diff
@@ -0,0 +1,38 @@
+---
+layout: default
+title: processexists
+published: true
+tags: [reference, process functions, functions, processexists, process, processes, ps]
+---
+
+[%CFEngine_function_prototype(regex)%]
+
+ **Description:** Return whether a process matches the given regular expression `regex`.
+
+This function searches for the given regular expression in the process
+table. Use `.*sherlock.*` to find all the processes that match
+`sherlock`. Use `.*\bsherlock\b.*` to exclude partial matches like
+`sherlock123` (`\b` matches a word boundary).
+
+[%CFEngine_function_attributes(regex)%]
+
+The returned data container is a list of key-value maps. Each one is
+guaranteed to have the key `pid` with the process ID. The key `line`
+will also be available with the raw process table contents.
+
+The process table is usually obtained with something like
+`ps -eo user,pid,ppid,pgid,%cpu,%mem,vsize,ni,rss,stat,nlwp,stime,time,args`
+but the exact data under the `line` key may change per platform and per CFEngine release.
+
+**Example:**
+
+    classes:
+      "holmes" expression => processexists(".*sherlock.*")
+
+Output:
+
+    [ { "pid": "2378", "line": "...the ps output here" }, ... ]
```

Comment:
Also, this example is wrong if I understand the function correctly. There is no returned data, just true/false.